reputation/plugin_bot.go

```diff
@@ -513,3 +517,14 @@ func userPresentInRepLog(userID int64, guildID int64, parsed *dcmd.Data) (found
 	}
 	return true, nil
 }
+
+// Checks if the thanks detection is allowed to be run in the given channel
+func isThanksDetectionAllowedInChannel(config *models.ReputationConfig, channelID int64) bool {
+	if len(config.WhitelistedThanksChannels) > 0 && common.ContainsInt64Slice(config.WhitelistedThanksChannels, channelID) {
+		return true
+	}
+	if len(config.BlacklistedThanksChannels) > 0 && common.ContainsInt64Slice(config.BlacklistedThanksChannels, channelID) {
+		return false
+	}
+	return len(config.WhitelistedThanksChannels) == 0 && len(config.BlacklistedThanksChannels) == 0
```

Comment:
Thanks for pointing out, updated the conditions.
Also, it was not intentional for both whitelist and blacklist channels to be required.

Reply:
this newest change is also a little wonky, because if you only set a whitelist channel, it still triggers in other channels

I'd suggest
```

	if len(config.WhitelistedThanksChannels) == 0 && len(config.BlacklistedThanksChannels) >= 0 {
		return true
	}
	return len(config.WhitelistedThanksChannels) == 0 && len(config.BlacklistedThanksChannels) == 0
```

Reply:
if whitelist channels are set, then it will return true only if the channel in which thanks is triggered is present in whitelist channels. For all cases it will return false.

Reply:
looks good